src/App/components/ProductList/ProductList.tsx

```diff
@@ -1,40 +1,104 @@
-                ({
-                    id: raw.id,
-                    title: raw.title
-                })));
-        };
-        fetch();
-    },        [])          // при пустом массиве коллбек вызывается
-    // только при появлении компонента в dom-дереве
-
-
-    return (
-            <ul>
-                {products.map(product => (
-                    <li key={product.id}>
-                        <Link to={`/user/${product.id}`}>
-                            {product.title}
-                        </Link>
-                    </li>))}
-            </ul>
+import Card from "../Card";
+import { Option } from "../Filter/Filter";
+import styles from "./ProductList.module.scss";
+
+
+/** Пропсы, который принимает компонент Button */
+export type ProductListProps = React.PropsWithChildren<{
+  /** Категории товаров, которые нужно вывести*/
+  filter: Array<Option>;
+  /** Ограничение на количество товаров*/
+  limit?: number;
+}>;
+
+//возвращает массив карточек, отфильтрованных по выбранным категориям, с заданным лимитом
+const ProductList: React.FC<ProductListProps> = ({ filter, limit }) => {
+  const [products, setProducts] = useState<Product[]>([]);
+
+  useEffect(() => {
+    getData();
+  }, [limit, filter]);
+
+  const fetch = async (address: string) => {
+    const result = await axios({
+      method: "get",
+      url: address,
+    });
+
+    /* //через конкат
+                setProducts(products.concat(result.data.map((raw: Product) =>
+                    ({
+                        id: raw.id,
+                        title: raw.title,
+                        category: raw.category,
+                        description: raw.description,
+                        image: raw.image,
+                        price: raw.price,
+                        rating: raw.rating
+                  }))));
+
+        */
+    setProducts(
+      result.data.map((raw: Product) => ({
+        id: raw.id,
+        title: raw.title,
+        category: raw.category,
+        description: raw.description,
+        image: raw.image,
+        price: raw.price,
+        rating: raw.rating,
+      }))
     );
+  };
+
+  const getData = () => {
+    if (filter.length !== 0) {
+      setProducts([]);
+      filter.map((category: Option) => {
+        fetch("https://fakestoreapi.com/products/category/" + category.value +"?limit=" + limit);
+      });
+      setProducts(products.splice(0, limit));
+    } else {
+      fetch("https://fakestoreapi.com/products?limit=" + limit);
+    }
+  };
+
+  const getProductCount = () => {
+    if (filter.length !== 0) {
+      filter.map((category: Option) => {
+        fetch("https://fakestoreapi.com/products/category/" + category.value);
+      });
+    } else {
+      fetch("https://fakestoreapi.com/products");
+    }
+    return;
+  };
+
+  let navigate = useNavigate();
+
+  return (
+    <div className={`${styles.product_list}`}>
+      {products?.map(
+        (product) =>
+          product && (
```

Comment:
кажется, ничего не мешает сделать так:
```suggestion
      {products.map(
        (product) =>
          (
```